cognite/neat/_rules/transformers/_converters.py

```diff
@@ -220,6 +224,140 @@ def _transform(self, rules: VerifiedRules) -> VerifiedRules:
         raise ValueError(f"Unsupported conversion from {type(rules)} to {self._out_cls}")
 
 
+_T_Entity = TypeVar("_T_Entity", bound=ClassEntity | ViewEntity)
+
+
+class ToExtension(RulesTransformer[DMSRules, DMSRules]):
+    def __init__(
+        self, new_model_id: DataModelIdentifier, prefix: str | None = None, mode: Literal["composition"] = "composition"
+    ):
+        self.new_model_id = DataModelId.load(new_model_id)
+        self.prefix = prefix
+        self.mode = mode
+
+    def transform(self, rules: DMSRules | OutRules[DMSRules]) -> JustRules[DMSRules]:
+        # Copy to ensure immutability
+        verified = self._to_rules(rules)
+        if self.prefix is None and verified.metadata.as_data_model_id() in COGNITE_MODELS:
+            raise NeatValueError(f"Prefix is required when extending {verified.metadata.as_data_model_id()}")
+        source_id = verified.metadata.as_data_model_id()
+
+        dump = verified.dump()
+        dump["metadata"]["schema_"] = SchemaCompleteness.partial.value
+        dump["metadata"]["space"] = self.new_model_id.space
+        dump["metadata"]["external_id"] = self.new_model_id.external_id
+        if self.new_model_id.version is not None:
+            dump["metadata"]["version"] = self.new_model_id.version
+        new_model = DMSRules.model_validate(DMSInputRules.load(dump).dump())
+
+        for prop in new_model.properties:
+            if prop.container and prop.container.space == self.new_model_id.space:
+                prop.container = ContainerEntity(
+                    space=source_id.space,
+                    externalId=prop.container.suffix,
+                )
+
+        if self.mode == "composition":
+            new_model.containers = None
+            for view in new_model.views:
+                view.implements = None
+
+        if source_id in COGNITE_MODELS:
+            # Remove CognitePrefixes.
+            for prop in new_model.properties:
+                prop.view = self._remove_cognite_prefix(prop.view)
+                prop.class_ = self._remove_cognite_prefix(prop.class_)
+                if isinstance(prop.value_type, ViewEntity):
+                    prop.value_type = self._remove_cognite_prefix(prop.value_type)
+            for view in new_model.views:
+                view.view = self._remove_cognite_prefix(view.view)
+                view.class_ = self._remove_cognite_prefix(view.class_)
+
+        return JustRules(new_model)
+
+    def _remove_cognite_prefix(self, entity: _T_Entity) -> _T_Entity:
+        new_suffix = entity.suffix.replace("Cognite", self.prefix or "")
```

Comment:
so if self.prefix is not provided you would remove `Cognite` , meaning L241 is not needed?

Reply:
Renamed it org_name to make it clearer